src/ApiCore/ResourceTemplate/RelativeResourceTemplate.php

```diff
@@ -0,0 +1,296 @@
+            assert($segment->getSegmentType() !== Segment::VARIABLE_SEGMENT);
+
+            if ($segment->getSegmentType() == Segment::DOUBLE_WILDCARD_SEGMENT) {
+                $pathPiecesForSegment = array_slice($pathPieces, $pathPiecesIndex, $doubleWildcardPieceCount);
+                $pathPiece = implode('/', $pathPiecesForSegment);
+                $pathPiecesIndex += $doubleWildcardPieceCount;
+            } else {
+                $pathPiece = $pathPieces[$pathPiecesIndex++];
+            }
+            if (!$segment->matches($pathPiece)) {
+                throw $this->matchException($path);
+            }
+
+            // If we have a valid key, add our $pathPiece to the $bindings array. Note that there
+            // may be multiple copies of the same $segmentKey. This is because a flattened variable
+            // segment can match multiple pieces from the path. We can add these to an array and
+            // collapse them all once the bindings are complete.
+            if (isset($segmentKey)) {
+                $bindings += [$segmentKey => []];
+                $bindings[$segmentKey][] = $pathPiece;
+            }
+        }
+
+        // It is possible that we have left over path pieces, which can occur if our template does
+        // not have a double wildcard. In that case, the match should fail.
+        if ($pathPiecesIndex !== count($pathPieces)) {
+            throw $this->matchException($path);
+        }
+
+        // Collapse the bindings from lists into strings
+        $collapsedBindings = [];
+        foreach ($bindings as $key => $boundPieces) {
+            $collapsedBindings[$key] = implode('/', $boundPieces);
+        }
+        return $collapsedBindings;
+    }
+
+    private function matchException($path)
+    {
+        return new ValidationException("Could not match path '$path' to template '$this'");
+    }
+
+    /**
+     * @param Segment[] $segments
+     * @return array[] A list of [string, Segment] tuples
+     */
+    private static function buildKeySegmentTuples($segments)
+    {
+        $keySegmentTuples = [];
+        $positionalArgumentCounter = 0;
+        foreach ($segments as $segment) {
+            switch ($segment->getSegmentType()) {
+                case Segment::WILDCARD_SEGMENT:
+                case Segment::DOUBLE_WILDCARD_SEGMENT:
+                    $positionalKey = "\$$positionalArgumentCounter";
+                    $positionalArgumentCounter++;
+                    $keySegmentTuples[] = [$positionalKey, $segment];
+                    break;
+                default:
+                    $keySegmentTuples[] = [$segment->getKey(), $segment];
+            }
+        }
+        return $keySegmentTuples;
+    }
+
+    /**
+     * @param array[] A list of [string, Segment] tuples
+     * @return array[] A list of [string, Segment] tuples
+     */
+    private static function flattenKeySegmentTuples($keySegmentTuples)
```

Comment:
ditto

Reply:
Fixed in https://github.com/googleapis/gax-php/pull/209